### ISSUE TITLE
Bug: button customId as Invalid Date and class not matching the actual raw data

### ISSUE THREAD

**Issue by USERSATOSHI** [AkaruiDevelopment/Zeneth#2]
### Bug Type

Classes

### reproducible code

```js
- create a button component 
- log the interactionData sent by interaction class in InteractionCreate event
```

### Version of Zeneth

v0.0.1

### Version of Node.js

v18 (LTS)

### Excepted Output

expected to get my button's customId in interaction.customId

### Current Output

interaction.customId returns undefined 

and interaction.data.customId returns Invalid Date

### Relevant log output

```shell
{"appPermissions":"140737488355327","applicationId":"1122837640325103646","channel":{"flags":0,"guildId":"781998221655277588","id":"789505361098833940","lastMessageId":"1123326040060739715","lastPinTimestamp":"2023-03-03T20:09:40+00:00","name":"teeeeeeeest","nsfw":false,"permissions":"140737488355327","position":1,"rateLimitPerUser":0,"topic":null,"type":0,"__priority":0},"channelId":"789505361098833940","data":{"customId":null,"componentType":2},"guildId":"781998221655277588","guildLocale":"en-US","id":"1123326045928571011","locale":"en-US","member":{"avatar":null,"deaf":false,"joinedAt":"2020-11-27T21:41:44.873Z","mute":false,"nick":"vhhhhhh","pending":false,"permissions":"140737488355327","roles":["782565291036770354"],"user":{"id":"555752666223869953","username":"wo6_","discriminator":0,"avatar":"85245324148931881444259375725465950031","bot":false,"system":false,"verified":false,"publicFlags":4194560,"globalName":"Ahmed.","guildIds":[],"__priority":0},"guildId":"781998221655277588"},"message":{"attachments":[],"author":{"id":"1122837640325103646","username":"MangaCord","discriminator":8866,"avatar":"133755183640875330805933793981530121754","bot":true,"system":false,"verified":false,"publicFlags":0,"guildIds":[],"__priority":0},"channelId":"789505361098833940","components":[{"type":1,"components":[{"type":2,"style":1,"label":null,"customId":null}]}],"content":"Pong!  288 ms","editedTimestamp":null,"embeds":[],"flags":0,"id":"1123326040060739715","pinned":false,"timestamp":"2023-06-27T18:56:38.339Z","tts":false,"type":0,"mentions":{"everyone":false,"roles":[],"users":[],"channels":[]},"__priority":0},"token":"","type":3,"version":1}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Asayukiii** [comment]
Same here with slash commands
```js
Interaction {
  appPermissions: '136758201539648',
  applicationId: 1077774805530591283n,
  channel: Channel {
    flags: 0,
    guildId: 966131185120059424n,
    id: 1092490880436408410n,
    lastMessageId: 1124113110115024947n,
    lastPinTimestamp: '2023-06-16T18:16:21+00:00',
    name: 'general',
    nsfw: false,
    permissions: '140737488355327',
    position: 0,
    rateLimitPerUser: 0,
    topic: null,
    type: 0,
    __priority: 0
  },
  channelId: 1092490880436408410n,
  data: { type: 1, name: Invalid Date, id: 1124136169052053526n },
  guildId: 966131185120059424n,
  guildLocale: 'en-US',
  id: 1124138367714611250n,
  locale: 'en-US',
  member: Member {
    avatar: null,
    deaf: false,
    joinedAt: 2023-06-12T01:10:15.531Z,
    mute: false,
    nick: null,
    pending: false,
    permissions: '140737488355327',
    roles: [ 999772685120852091n ],
    user: User {
      id: 1077733015846932530n,
      username: 'erichanuwu',
      discriminator: 0,
      avatar: 48493756390399700392297774169516947974n,
      bot: false,
      system: false,
      verified: false,
      publicFlags: 0,
      globalName: 'Erisa',
      guildIds: [],
      __priority: 0
    },
    guildId: 966131185120059424n
  },
  token: 'aW50ZXJhY3Rpb246MTEyNDEzODM2NzcxNDYxMTI1MDpaNlNsSDQ3S0Y3dG44R2Vma01JWHppSFFTWmhIeE5WM0F6YU9DV2daRTZjSk53SEFQNV...',
  type: 2,
  version: 1
}
```

**Comment by USERSATOSHI** [comment]
fixed with 650b3e5 commit @Airycita
